### PR TITLE
Luis/persistency scheme

### PR DESCRIPTION
**Pull request by LDiazN**
# Persistency Scheme
## Problem Description
We need a way to persist data computed with out library in some persistent storage, and it should be flexible enough so the user can plug & play his custom persistency methods to store the data as he needs. 

## Proposed solution
Create a new base class for storage manager objects that can be easily overridden.  We propose a class with common operations to perform over the stored data: 
* `get_matching( ScrapedData -> Bool ) -> [ScrapedData]` - Given a predicate, retrieve all objects stored matching such predicate
* `filter_scraped_urls([str]) -> List[str]` - Filter out urls already scraped in the given list, leaving only urls that are to be scraped.
* `was_scraped(str) -> bool` - tells if the given url is already scraped (there's no need to scrape it for new data)
* `save([ScrapedData])` - save provided data instances to disk
* `delete([str])` - delete data related to provided urls from disk

Also, we provide a json-based implementation for local storage and testing purposes.

## Important changed:
* add Base Class for some other persistency manager objects
* add Json Based implementation

## How to test it 

import `JsonManager` class: 
```
from c4v.scraper.persistency_manager.json_storage_manager import JsonManager
```

Create an instance of such class
```
jman = JsonManager("path/to/json_file/in/your/computer.json")
```

use this object to test its corresponding methods

## Relevant files:
* `c4v/scraper/persistency_manager/base_persistency_manager.py` - Base persistency manager class implementation
* `c4v/scraper/persistency_manager/json_storage_manager.py`  - Json based implementation 

## Further work 
* Write a more efficient implementation of the json manager
* create a file manager class, so we can use it instead of manually opening the file. This way we can create fake files to improve testability for the json manager and further new classes that will try to work with files
* Some Idea about some other common operations that may be part of a persistency manager?
* Include current work in CLI utility to save scraped data into disk

### CODE REVIEW

File: src/c4v/scraper/persistency_manager/base_persistency_manager.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+# Local imports
+from c4v.scraper.scraped_data_classes.scraped_data import ScrapedData
+
+
+class BasePersistencyManager:
+    """
+        Base class to provide support for persistency management
+    """
+
+    def get_matching(
```
Comment (Edilmo) [inline src/c4v/scraper/persistency_manager/base_persistency_manager.py:22]: What is the purpose of this API?

This is a search, more concretely, it's equality based search which I don't think we need and it's just too expensive to have.
If we need search capabilities, the API should reflect the boundaries of a realistic search.

File: src/c4v/scraper/persistency_manager/json_storage_manager.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
+"""
```
Comment (Edilmo) [inline src/c4v/scraper/persistency_manager/json_storage_manager.py:1]: Is this our local parallel implementation to the BigQuery one?

if that is the case, then this is not a good option.
Can we change this by an implementation based on SQLite please: https://docs.python.org/3/library/sqlite3.html

File: tests/scraper/scraped_data_test.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import dataclasses
```
Comment (Edilmo) [inline tests/scraper/scraped_data_test.py:1]: test files should start with the test prefix and not end with a test suffix.

**Review by Edilmo — APPROVED** [review]
Approving to unblock, but there are 2 important changes that must be done. I think they should be done in this same PR.

File: src/c4v/scraper/persistency_manager/json_storage_manager.py (hunk 2 above)
Comment (LDiazN) [reply src/c4v/scraper/persistency_manager/json_storage_manager.py:1]: This is for our local parallel implementation, to do local testing, but I'll agree that it's inefficient, I'll switch it to an SQLite based one.

**Review by dieko95 — APPROVED** [review]

**Pull request merged**